File: src/scripts/Levels.cs
using System.Collections.Generic;
using UnityEngine;

public static class Levels
{
    public static IList<Placement> Layout = new List<Placement>();

    static Levels()
    {
        var level = new Placement();

        level.AddBlock(0, 1);
        level.AddBlock(1, 0);
        level.AddTarget(0, 0);
        Layout.Add(level);

        level = new Placement();
        level.AddBlock(0, 1);
        level.AddBlock(1, 0);
        level.AddBlock(2, 1);
        level.AddTarget(1, 1);
        Layout.Add(level);

        level = new Placement();
        level.AddBlock(0, 1);
        level.AddBlock(1, 0);
        level.AddBlock(2, 1);
        level.AddBlock(1, 2);
        level.AddTarget(1, 1);
        Layout.Add(level);

        level = new Placement();
        level.AddBlock(0, 2);
        level.AddBlock(1, 1);
        level.AddBlock(1, 0);
        level.AddTarget(0, 0);
        Layout.Add(level);

        level = new Placement();
        level.AddBlock(0, 1);
        level.AddBlock(1, 0);
        level.AddBlock(1, 1);
        level.AddBlock(2, 0);
        level.AddTarget(1, 0);
        Layout.Add(level);

        level = new Placement();
        // gen 1
        level.AddBlock(0, 0);
        level.AddBlock(2, 1);
        level.AddBlock(0, 2);
        level.AddBlock(0, 1);
        level.AddBlock(1, 3);
        level.AddBlock(1, 2);
        level.AddBlock(1, 1);
        level.AddTarget(0, 1);
        Layout.Add(level);

        level = new Placement();
        level.AddBlocks(new Vector2Int(0, 0), new Vector2Int(3, 3));
        level.RemoveBlock(1, 2);
        level.RemoveBlock(0, 0);
        level.RemoveBlock(3, 3);
        level.RemoveBlock(3, 0);
        level.RemoveBlock(0, 3);
        level.RemoveBlock(2, 1);
        level.AddTarget(1, 1);
        Layout.Add(level);

        level = new Placement();
        level.AddBlocks(new Vector2Int(0, 0), new Vector2Int(3, 3));
        level.RemoveBlock(1, 2);
        level.RemoveBlock(3, 3);
        level.RemoveBlock(1, 1);
        level.RemoveBlock(0, 3);
        level.RemoveBlock(3, 2);
        level.RemoveBlock(2, 1);
        level.RemoveBlock(1, 1);
        level.RemoveBlock(0, 0);
        level.AddTarget(3, 1);
        Layout.Add(level);

        level = new Placement();
        level.AddBlocks(new Vector2Int(0, 0), new Vector2Int(7, 7));
        level.RemoveBlock(0, 0);
        level.RemoveBlock(7, 7);
        level.RemoveBlock(7, 0);
        level.RemoveBlock(0, 7);
        level.RemoveBlock(4, 3);
        level.RemoveBlock(3, 4);
        level.AddTarget(3, 3);
        Layout.Add(level);
    }
}

File: src/Monodes/levels.ts
import LevelBuilder from './LevelBuilder';
import Level from './Level';

const builder = new LevelBuilder();

const levels: Level[] = [];

builder.addBlock(0, 1);
builder.addBlock(1, 0);
builder.addTarget(0, 0);
levels.push(builder.constructLevel());

builder.addBlock(0, 1);
builder.addBlock(1, 0);
builder.addBlock(2, 1);
builder.addTarget(1, 1);
levels.push(builder.constructLevel());

builder.addBlock(0, 1);
builder.addBlock(1, 0);
builder.addBlock(2, 1);
builder.addBlock(1, 2);
builder.addTarget(1, 1);
levels.push(builder.constructLevel());

builder.addBlock(0, 2);
builder.addBlock(1, 1);
builder.addBlock(1, 0);
builder.addTarget(0, 0);
levels.push(builder.constructLevel());

builder.addBlock(0, 1);
builder.addBlock(1, 0);
builder.addBlock(1, 1);
builder.addBlock(2, 0);
builder.addTarget(1, 0);
levels.push(builder.constructLevel());

builder.addBlock(0, 0);
builder.addBlock(2, 1);
builder.addBlock(0, 2);
builder.addBlock(0, 1);
builder.addBlock(1, 3);
builder.addBlock(1, 2);
builder.addBlock(1, 1);
builder.addTarget(0, 1);
levels.push(builder.constructLevel());

builder.addBlocks({ x: 0, y: 0 }, { x: 3, y: 3 });
builder.removeBlock(1, 2);
builder.removeBlock(0, 0);
builder.removeBlock(3, 3);
builder.removeBlock(3, 0);
builder.removeBlock(0, 3);
builder.removeBlock(2, 1);
builder.addTarget(1, 1);
levels.push(builder.constructLevel());

builder.addBlocks({ x: 0, y: 0 }, { x: 3, y: 3 });
builder.removeBlock(1, 2);
builder.removeBlock(3, 3);
builder.removeBlock(1, 1);
builder.removeBlock(0, 3);
builder.removeBlock(3, 2);
builder.removeBlock(2, 1);
builder.removeBlock(1, 1);
builder.removeBlock(0, 0);
builder.addTarget(3, 1);
levels.push(builder.constructLevel());

builder.addBlocks({ x: 0, y: 0 }, { x: 7, y: 7 });
builder.removeBlock(0, 0);
builder.removeBlock(7, 7);
builder.removeBlock(7, 0);
builder.removeBlock(0, 7);
builder.removeBlock(4, 3);
builder.removeBlock(3, 4);
builder.addTarget(3, 3);
levels.push(builder.constructLevel());

export default levels;
